src/machine.rs

```diff
@@ -93,17 +93,21 @@ pub struct Evaluator<'tcx> {
 
     /// TLS state.
     pub(crate) tls: TlsData<'tcx>,
+
+    /// If enabled, the `env_vars` field is populated with the host env vars during initialization.
+    pub(crate) communicate: bool,
 }
 
 impl<'tcx> Evaluator<'tcx> {
-    pub(crate) fn new() -> Self {
+    pub(crate) fn new(communicate: bool) -> Self {
         Evaluator {
-            env_vars: HashMap::default(),
+            env_vars: EnvVars::default(),
```

Comment:
Instead of using `default` and then later calling `init`, please call `EnvVars::new(communicate)` here and then that constructor can figure it out either way.

Reply:
That was my first thought. However, I couldn't figure out how to do it properly because we would need to access `Memory` (which is initialized inside `Interpcx::new`) before `Evaluator` is initialized (which is done before calling `Interpcx::new` in `create_ecx`). One solution would be to change `create_ecx` to receive `Memory` by parameter instead.

Reply:
Hm, good point. Add a comment here explaining that.

Reply:
Fixme? or just a comment?

Reply:
Not a FIXME; just explaining why things aren't nicer.